docs/fsharp-cheatsheet.md

```diff
@@ -740,6 +743,112 @@ Another way of implementing interfaces is to use *object expressions*.
 
 *Partial active patterns* share the syntax of parameterized patterns but their active recognizers accept only one argument.
 
+<div id="asynchronous-programming"></div>
+
+## Asynchronous Programming
+
+F# asynchronous programming is centered around two core concepts: async computations and tasks.
+
+    async {
+        do! Async.Sleep (waitInSeconds * 1000)
+        let! asyncResult = asyncComputation
+        use! disposableResult = iDisposableAsyncComputation
+    }
+
+### Async vs Tasks
+
+An async computation is a unit of work, and a task is a promise of a result. A subtle but important distinction.
+Async computations are composable and are not started until explicitly requested; Tasks (when created using the `task` expression) are _hot_:
```

Comment:
I think it's important to mention that async exprs flow cancellation tokens implicitly.
The big thing you are not conveying here is that while .NET and C# do use Task as the basic atom, and it can be more efficient, that efficiency is in the main irrelevant in the context of actual IO bound work. I'll make reference to specific places where you send a bad steer later...